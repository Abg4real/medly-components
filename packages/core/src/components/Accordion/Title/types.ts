import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp{
    color?: string;
    isActive?: boolean;
    onToggle?: () => void;
    bgColor?: string;

}


