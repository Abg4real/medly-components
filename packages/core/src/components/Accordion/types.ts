import { Props as TitleProps } from './Title';

export interface Props {
    Title?: React.SFC<TitleProps>;
    Content?: React.SFC;
}