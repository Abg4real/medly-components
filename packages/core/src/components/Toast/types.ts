import { HTMLProps, WithThemeProp } from '@medly-components/utils';

export type ToastVariants = 'success' | 'error' | 'info' | 'warning';

export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'id'> {
    /** Id of the Toast */
    id?: number;
    /** Toast purpose */
    variant: ToastVariants;
    /** Icon to be shown in the toast */
    icon?: React.SFC<any>;
    /** Toast Header */
    header?: string;
    /** Toast message */
    message?: string;
}

export interface StyledProps extends WithThemeProp, Props {
    show?: boolean;
}
