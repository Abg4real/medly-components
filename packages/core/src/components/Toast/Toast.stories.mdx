import { Toast } from './Toast';
import { defaultTheme } from '@medly-components/theme';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Toast.stories';

<Meta title="Core/Toast" component={Toast} />

# Toast

### Basic Toast

<Preview withToolbar>
    <Story name="Basic">{stories.Basic}</Story>
</Preview>

### Props

<Props of={Toast} />

### ToastTheme

<Props of={stories.ThemeInterface} />

### ToastTheme Defaults

<code>{JSON.stringify(defaultTheme.toast, null, 2)}</code>
