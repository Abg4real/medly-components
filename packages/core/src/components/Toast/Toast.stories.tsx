import { defaultTheme, ToastTheme } from '@medly-components/theme';
import { styled } from '@medly-components/utils';
import React from 'react';
import { ToastProps } from './types';

export const variants: ToastProps['variant'][] = ['error', 'warning', 'success', 'info'];

export const ThemeInterface: React.SFC<ToastTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.toast
};

export const Div = styled.div`
    display: flex;
    flex-direction: column;
    & > * {
        margin-bottom: 2rem;
    }
`;
