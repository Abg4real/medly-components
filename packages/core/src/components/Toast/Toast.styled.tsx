import { SvgIcon } from '@medly-components/icons';
import { styled } from '@medly-components/utils';
import Text from '../Text';
import { StyledProps } from './types';

// @ts-ignore
const getColor = ({ theme, variant }: StyledProps) => theme.toast[`${variant}Color`];

export const Toast = styled('div')<StyledProps>`
    border-radius: 5px;
    display: flex;
    align-items: center;
    min-height: 50px;
    max-height: 50px;
    min-width: 250px;
    justify-content: space-between;
    padding: 0 20px;
    margin: 5px;
    opacity: ${({ show }) => (show ? 1 : 0)};
    transition: opacity 0.3s ease;

    background-color: ${getColor};

    ${Text.Style} {
        color: ${({ theme }) => theme.toast.textColor};
        margin: 0px 10px;
    }
    ${SvgIcon} {
        * {
            fill: ${({ theme }) => theme.toast.textColor};
        }
    }
`;
