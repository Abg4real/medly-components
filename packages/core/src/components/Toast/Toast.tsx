import { ClearIcon, DoneIcon, ErrorIcon, NotificationsIcon, WarningAmberIcon } from '@medly-components/icons';
import { WithStyle } from '@medly-components/utils';
import React, { SFC, useCallback } from 'react';
import Text from '../Text';
import { removeToast } from '../ToastContainer/ToastStore';
import * as Styled from './Toast.styled';
import { Props } from './types';

export const Toast: SFC<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => {
        const { id, variant, header, icon: Icon, message, ...restProps } = props;

        const handleClose = useCallback(() => removeToast(id), [id]);

        return (
            <Styled.Toast ref={ref} variant={variant} {...restProps}>
                <Styled.SvgWrapper variant={variant}>
                    {Icon ? (
                        <Icon />
                    ) : (
                        {
                            error: <ErrorIcon />,
                            warning: <WarningAmberIcon />,
                            info: <NotificationsIcon />,
                            success: <DoneIcon />
                        }[variant]
                    )}
                </Styled.SvgWrapper>
                <Styled.ToastContent>
                    {header && <Text textWeight="Medium">{header}</Text>}
                    {message && <Text fullWidth>{message}</Text>}
                </Styled.ToastContent>
                <ClearIcon size="M" onClick={handleClose} />
            </Styled.Toast>
        );
    })
);
Toast.displayName = 'Toast';
Toast.Style = Styled.Toast;
