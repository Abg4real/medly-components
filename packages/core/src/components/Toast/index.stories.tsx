import { action } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { Toast } from './Toast';
import { Props } from './types';

const variant: Array<Props['variant']> = ['error', 'warning', 'success', 'info'];

storiesOf('Core', module).add('Toast', () => (
    <Toast variant={select('Variant', variant, 'success')} onClose={action('Close icon cliked')}>
        Hello
    </Toast>
));
