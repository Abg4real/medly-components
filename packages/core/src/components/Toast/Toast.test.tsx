import { TestUtils } from '@medly-components/utils';
import React from 'react';
import { Toast } from './Toast';
import { Props } from './types';

const { render, cleanup, fireEvent } = TestUtils;

describe('Toast', () => {
    afterEach(cleanup);

    test.each(['error', 'warning', 'success', 'info'])('should render properly with %s variant', (variant: Props['variant']) => {
        const { container } = render(<Toast variant={variant}>Hello</Toast>);
        expect(container).toMatchSnapshot();
    });

    it('should render properly if child jsx', () => {
        const { container } = render(
            <Toast variant="success">
                <p>Hello</p>
            </Toast>
        );
        expect(container).toMatchSnapshot();
    });

    it('should call on Close when click on the clear icon', () => {
        const mockOnClose = jest.fn();
        const { container } = render(
            <Toast variant="success" onClose={mockOnClose}>
                Hello
            </Toast>
        );
        fireEvent.click(container.querySelectorAll('svg')[1]);
        expect(mockOnClose).toBeCalled();
    });
});
