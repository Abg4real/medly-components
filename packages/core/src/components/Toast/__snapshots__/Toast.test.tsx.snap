// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render properly if child jsx 1`] = `
.c2 {
  font-size: 16px;
  cursor: inherit;
}

.c2 * {
  fill: #616161;
}

.c3 .c1 * {
  fill: #ffffff;
}

.c0 {
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50px;
  max-height: 50px;
  min-width: 250px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 20px;
  margin: 5px;
  opacity: 1;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  background-color: #008000;
}

.c0 .sc-AykKF {
  color: #ffffff;
  margin: 0px 10px;
}

.c0 .c1 * {
  fill: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        fill="currentColor"
      />
    </svg>
    <p>
      Hello
    </p>
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Toast should render properly with error variant 1`] = `
.c2 {
  font-size: 16px;
  cursor: inherit;
}

.c2 * {
  fill: #616161;
}

.c4 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c5 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c5 .c1 * {
  fill: #ffffff;
}

.c0 {
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50px;
  max-height: 50px;
  min-width: 250px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 20px;
  margin: 5px;
  opacity: 1;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  background-color: red;
}

.c0 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c0 .c1 * {
  fill: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M511.988 288.9c-.478 17.43-15.217 31.1-32.653 31.1H424v16c0 21.864-4.882 42.584-13.6 61.145l60.228 60.228c12.496 12.497 12.496 32.758 0 45.255-12.498 12.497-32.759 12.496-45.256 0l-54.736-54.736C345.886 467.965 314.351 480 280 480V236c0-6.627-5.373-12-12-12h-24c-6.627 0-12 5.373-12 12v244c-34.351 0-65.886-12.035-90.636-32.108l-54.736 54.736c-12.498 12.497-32.759 12.496-45.256 0-12.496-12.497-12.496-32.758 0-45.255l60.228-60.228C92.882 378.584 88 357.864 88 336v-16H32.666C15.23 320 .491 306.33.013 288.9-.484 270.816 14.028 256 32 256h56v-58.745l-46.628-46.628c-12.496-12.497-12.496-32.758 0-45.255 12.498-12.497 32.758-12.497 45.256 0L141.255 160h229.489l54.627-54.627c12.498-12.497 32.758-12.497 45.256 0 12.496 12.497 12.496 32.758 0 45.255L424 197.255V256h56c17.972 0 32.484 14.816 31.988 32.9zM257 0c-61.856 0-112 50.144-112 112h224C369 50.144 318.856 0 257 0z"
        fill="currentColor"
      />
    </svg>
    <span
      class="c3 c4"
    >
      Hello
    </span>
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Toast should render properly with info variant 1`] = `
.c2 {
  font-size: 16px;
  cursor: inherit;
}

.c2 * {
  fill: #616161;
}

.c4 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c5 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c5 .c1 * {
  fill: #ffffff;
}

.c0 {
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50px;
  max-height: 50px;
  min-width: 250px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 20px;
  margin: 5px;
  opacity: 1;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  background-color: #6BA3E5;
}

.c0 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c0 .c1 * {
  fill: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        fill="currentColor"
      />
    </svg>
    <span
      class="c3 c4"
    >
      Hello
    </span>
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Toast should render properly with success variant 1`] = `
.c2 {
  font-size: 16px;
  cursor: inherit;
}

.c2 * {
  fill: #616161;
}

.c4 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c5 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c5 .c1 * {
  fill: #ffffff;
}

.c0 {
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50px;
  max-height: 50px;
  min-width: 250px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 20px;
  margin: 5px;
  opacity: 1;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  background-color: #008000;
}

.c0 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c0 .c1 * {
  fill: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        fill="currentColor"
      />
    </svg>
    <span
      class="c3 c4"
    >
      Hello
    </span>
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;

exports[`Toast should render properly with warning variant 1`] = `
.c2 {
  font-size: 16px;
  cursor: inherit;
}

.c2 * {
  fill: #616161;
}

.c4 {
  font-size: 14px;
  font-weight: 400;
  color: inherit;
}

.c5 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c5 .c1 * {
  fill: #ffffff;
}

.c0 {
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 50px;
  max-height: 50px;
  min-width: 250px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 20px;
  margin: 5px;
  opacity: 1;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  background-color: #FBD94F;
}

.c0 .c3 {
  color: #ffffff;
  margin: 0px 10px;
}

.c0 .c1 * {
  fill: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 576 512"
      width="1em"
    >
      <path
        d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        fill="currentColor"
      />
    </svg>
    <span
      class="c3 c4"
    >
      Hello
    </span>
    <svg
      class="c1 c2"
      height="1em"
      viewBox="0 0 352 512"
      width="1em"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
